src/uint/modular/constant_mod.rs

```diff
@@ -99,6 +99,29 @@ impl<MOD: ResidueParams<LIMBS>, const LIMBS: usize> Residue<MOD, LIMBS> {
         }
     }
 
+    /// Instantiates a new `Residue` that represents this `integer` mod `MOD`.
+    /// If the modulus represented by `MOD` is not odd, this function will panic; use [`new_checked`][`Residue::new_checked`] if you want to be able to detect an invalid modulus.
+    pub const fn new(integer: &Uint<LIMBS>) -> Self {
+        // A valid modulus must be odd
+        if MOD::MODULUS.ct_is_odd().to_u8() == 0 {
+            panic!("modulus must be odd");
+        }
```

Comment:
Absent inline const (which might be worth leaving a TODO comment about) this will force this check into a const context, evaluating it at compile time:

```suggestion
        const _: () = if MOD::MODULUS.ct_is_odd().to_u8() == 0 {
            panic!("modulus must be odd");
        };
```

...and with a compile-time check as to the validity of the modulus, I don't think we need `new_checked` since there will be no panic condition.

Reply:
Aah, unfortunate